internal/graph/graph_test.go

```diff
@@ -0,0 +1,546 @@
+			expectedIngresses: []*RelationshipIngress{
+				{
+					Type:    DirectIngress,
+					Ingress: typesystem.RelationReference("group", "member"),
+				},
+			},
+		},
+		{
+			name: "No graph relationship connectivity",
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "team",
+						Relations: map[string]*openfgapb.Userset{
+							"member": typesystem.This(),
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"member": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										typesystem.RelationReference("team", "member"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			targetUserRef:     typesystem.RelationReference("user", ""),
+			targetObjectRef:   typesystem.RelationReference("team", "member"),
+			expectedIngresses: []*RelationshipIngress{},
+		},
+		{
+			name: "",
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "group",
+						Relations: map[string]*openfgapb.Userset{
+							"member": typesystem.This(),
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"member": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										typesystem.RelationReference("user", ""),
+									},
+								},
+							},
+						},
+					},
+					{
+						Type: "folder",
+						Relations: map[string]*openfgapb.Userset{
+							"viewer": typesystem.This(),
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"viewer": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										typesystem.RelationReference("user", ""),
+										typesystem.RelationReference("group", "member"),
+									},
+								},
+							},
+						},
+					},
+					{
+						Type: "document",
+						Relations: map[string]*openfgapb.Userset{
+							"parent": typesystem.This(),
+							"viewer": typesystem.TupleToUserset("parent", "viewer"),
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"parent": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										typesystem.RelationReference("folder", ""),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			targetUserRef:   typesystem.RelationReference("user", ""),
+			targetObjectRef: typesystem.RelationReference("document", "viewer"),
+			expectedIngresses: []*RelationshipIngress{
+				{
+					Type:    DirectIngress,
+					Ingress: typesystem.RelationReference("folder", "viewer"),
+				},
+				{
+					Type:    DirectIngress,
+					Ingress: typesystem.RelationReference("group", "member"),
+				},
+			},
+		},
+		{
+			name: "",
```

Comment:
Not sure the best name for this? Any recommendations 😄 ?

Reply:
Test1!